### DISCUSSION TITLE
GPIO configure as dimmer

### DISCUSSION THREAD

**Discussion by Danit2** [ptvoinfo/zigbee-configurable-firmware#121] [Q&A]
First of all, I would like to say thank you for this great project. I really like it.

But now i have a question?

Is it possible to configure an input as a dimmer with on and off function.

I would do it like this:
On the first input, a short press will send an 'on' command and a long press a 'move level up' command. On the second input, a short press will send an 'off' command and a long press a 'move level down' command.

I would like to bind this directly to a light.
So it must be possible to bind a LevelControl Cluster an a OnOff Cluster from the same endpoint to a light. 

Is this possible or not?

Can you explain to me how to configure this?

I hope you understand what i will do. 

Thanks.

**Comment by ptvoinfo** [comment]
Sorry, but you cannot send different commands from the same button.

**Reply by Danit2** [reply]
So this is not possible? 
Do you plan to integrate this?

Then i can use your firmware as a dimmer switch. 

What is the function of the option `dimm up with off` and `dimm down with off`when you configure a input as GPIO? 

Thanks.

**Reply by ptvoinfo** [reply]
`dimm up with off` it is the Zigbee command. The remote device can handle this command and disable an output after a some level.

**Reply by Danit2** [reply]
Ahh ok i see.
But do you think you can implement a real dimmer function in your next version?
This would be very great.

Nice would be a single dimmer button and a double dimmer button. 

- dimmer_single: Use one push button to toggle the state of a light and also to control its level. A short press will send a 'toggle' command and long presses will move the level up and down alternately.

- dimmer_double: Use two push buttons to switch a light on and off and also to control its level. On the first input, a short press will send an 'on' command and a long press a 'move level up' command. On the second input, a short press will send an 'off' command and a long press a 'move level down' command.

I give you a donate for that when you want.
